### ISSUE TITLE
 quote unsupported punctuation on `?;`

### ISSUE THREAD

**Issue by ModProg** [rune-rs/rune#580]
Probably quote shouldn't parse any joint punctuation as one token.

**Comment by udoprog** [comment]
Could you elaborate? I don't know what this issue relates to.

**Comment by ModProg** [comment]
`quote!(something?;)` throws this error.

**Comment by udoprog** [comment]
Can you reproduce the exact error here for reference?

**Comment by ModProg** [comment]
```rustc
error: unsupported punctuation
  --> src/env.rs:47:30
   |
47 |         rune::macros::quote!(?;);
   |                              ^
```

**Comment by ModProg** [comment]
I expect the reason is that `quote` parses all punctuation that is `Spacing::Joint` as one tt, but it should be two in this case.

**Comment by udoprog** [comment]
Right, so yeah, it's an error we raise: https://github.com/rune-rs/rune/blob/40786fe4ec1f004cb443146032fbade9974111ca/crates/rune-macros/src/quote.rs#L115
So we need to fix it, somehow :)

**Comment by udoprog** [comment]
So a way to do this would be to improve `kind_from_punct` to process more than one punctuation in case it's a joint composite which can actually be decomposed into multiple punctuations. Since both `?` and `;` [are already separate punctuations](https://github.com/rune-rs/rune/blob/main/assets/tokens.yaml#L297).

**Comment by ModProg** [comment]
I have a list of the conditions to parse rust tts from joint tokens here:
https://github.com/ModProg/proc-macro-utils/blob/196525b066a3b5f8d751da6e7574731790827bb3/src/parser.rs#L1047
